### ISSUE TITLE
Output streamflow for catchments with less than 2 nexus

### ISSUE THREAD

**Issue by lcunha0118** [NOAA-OWP/ngen#428]
This is an issue that came up with the calibration of CAMELS basin, but I believe it should be solved at Ngen not only in the calibration code. When the area of interest only contains 1 nexus (headwater or group of basins that drain to one unique nexus), there is nothing to be routed and t-route cannot be ran, which is okay. However, we should still produce streamflow as output for that run. Streamflow in that case would be the weighted average runoff from all catchments times the basin area.    

## Current behavior
Ngen produces runoff output for headwater catchments but not streamflow output. 

## Expected behavior
Ngen should produce streamflow as output independently of how many catchments and nexus are in the hydrofabrics.

**Comment by mattw-nws** [comment]
Yeah this is an ngen/t-route issue... we have the per-catchment contributions to the nexus, and that is what we should probably be using for the catchment streamflow (vs. weighted average based on basin area)... it's just that we don't run the routing code on the headwater catchments so it never gets recorded. We'll add this to a list of things to do with Inland Hydraulics.
